android/jni/com/mapswithme/maps/SearchEngine.cpp

```diff
@@ -240,21 +240,14 @@ extern "C"
   {
     lock_guard<mutex> guard(g_resultsMutex);
     Result const & result = g_results.GetResult(index);
-    g_framework->PostDrapeTask([result]()
-    {
-      g_framework->NativeFramework()->ShowSearchResult(result);
-    });
+    g_framework->NativeFramework()->ShowSearchResult(result);
```

Comment:
I guess it's not a good idea to lock a mutex and then call quite heavy Framework::ShowSearchResult.